Merged the two classes into one since they do they exact same thing.

satoon101 opened the pull request.

satoon101 committed:
Merged the two classes into one since they do they exact same thing.

Moved the checking and super call into  which makes more sense.

Predz commented:
Don't know why I didn't pick up on that sooner, but thanks again Satoon. I will test the changes in a bit, but I trust they will work anyhow. :)

ThomasVieth committed:
Changed variable names along with class changes

viewmodel to be renamed model in the methods to better represent the data being parsed.
